### PR TITLE
fixes #20193 - upgrade storybook to 3.1

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

File: webpack/stories/index.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React from 'react';
 
 /* eslint-disable no-unused-vars */
```
Comment (tbrisker) [inline webpack/stories/index.js]: is this comment needed? I don't see any unused vars (I think linkTo was the unused one?)

File: package.json [hunk 2]
```diff
@@ -76,7 +80,7 @@
     "lint": "./node_modules/.bin/eslint -c .eslintrc webpack/ || exit 0",
     "test": "node node_modules/.bin/jest",
     "test:watch": "node node_modules/.bin/jest --watchAll",
-    "storybook": "start-storybook --dont-track -p 6006",
```
Comment (tbrisker) [inline package.json:79]: does this mean that tracking is enabled now or did the default change?

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+          'plugins': [
+            path.join(__dirname, '..', 'node_modules/babel-plugin-transform-object-rest-spread'),
+            path.join(__dirname, '..', 'node_modules/babel-plugin-transform-object-assign')
+          ]
+        }
       },
       {
         test: /\.css$/,
-        loader: 'style-loader!css-loader'
+        loaders: ['style-loader', 'css-loader', 'sass-loader']
```
Comment (tbrisker) [inline .storybook/webpack.config.js]: I think `sass-loader` shouldn't run on `.css` files, only on `.scss`

File: package.json [hunk 4]
```diff
@@ -5,7 +5,12 @@
   "description": "Foreman isn't really a node module, these are just dependencies needed to build the webpack bundle. 'dependencies' are the asset libraries in use and 'devDependencies' are used for the build process.",
   "private": true,
   "devDependencies": {
-    "@kadira/storybook": "^2.5.2",
+    "@storybook/addon-actions": "^3.1.2",
+    "@storybook/addon-info": "^3.1.4",
+    "@storybook/addon-knobs": "3.1.2",
```
Comment (tbrisker) [inline package.json]: Is there a reason this one is locked to a specific version and the other addons are not? 
Also, do we actually use any of the addons other then actions anywhere?

**Review by tbrisker — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (ohadlevy) [reply package.json:79]: sadly that option has been removed, I was unable to find if it actually tracks or not, based on my browser connections it seems like it does not.

**Comment by ohadlevy** [comment]
@tbrisker updated.

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
not sure why, but looks like tests broke completely?

**Comment by tbrisker** [comment]
Ok, looks like tests broke due to #4625, not related to this, opened #4644 to make sure js tests run when webpack config changes. Merging.

**Comment by tbrisker** [comment]
Thanks @ohadlevy !

**Pull request merged**